packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -249,10 +249,64 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
             // ODSP document ids are random guids (different per session)
             // fix the branch name in attributes
             // this prevents issues when generating summaries
-            const documentAttributes: api.IDocumentAttributes = JSON.parse(fromBase64ToUtf8(blob.content));
-            documentAttributes.branch = this.documentId;
+            if (blob.encoding === "base64") {
+                const documentAttributes: api.IDocumentAttributes = JSON.parse(fromBase64ToUtf8(blob.content));
+                documentAttributes.branch = this.documentId;
 
-            blob.content = fromUtf8ToBase64(JSON.stringify(documentAttributes));
+                blob.content = fromUtf8ToBase64(JSON.stringify(documentAttributes));
+            }
+            else {
+                const documentAttributes: api.IDocumentAttributes = JSON.parse(blob.content);
+                documentAttributes.branch = this.documentId;
+
+                blob.content = fromUtf8ToBase64(JSON.stringify(documentAttributes));
+            }
+        }
+
+        return blob.content;
+    }
+
+    public async readString(blobid: string): Promise<string> {
```

Comment:
There's a lot of duplicated code here between read and readString, consider if there's an opportunity to consolidate.